### PR TITLE
kubectl-gadget: init at 0.14.0

### PR DESCRIPTION
**Pull request by kranurag7**
###### Description of changes
kubectl-gadget is a collection of gadgets that allow you to troubleshoot and get insights from your kubernetes cluster using eBPF. 

https://github.com/inspektor-gadget/inspektor-gadget/tree/main/cmd/kubectl-gadget

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/kubectl-gadget/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  version = "0.14.0";
+
+  src = fetchFromGitHub {
+    owner = "inspektor-gadget";
+    repo = "inspektor-gadget";
+    rev = "v${version}";
+    sha256 = "sha256:16i9biyvzkpgxyfb41afaarnlm59vy02nspln5zq69prg6mp8rwa";
+  };
+
+  vendorSha256 = "sha256-Kj8gP5393++nPeX38TX6duB9OO/ql7hpRA5gTTtTl+M=";
```
Comment (wegank) [inline pkgs/applications/networking/cluster/kubectl-gadget/default.nix:14]: Please use `hash` and `vendorHash` whenever possible.

File: pkgs/applications/networking/cluster/kubectl-gadget/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "-X main.version=v${version}"
+  ];
+
+  subPackages = [ "cmd/kubectl-gadget" ];
+
+  meta = with lib; {
+    description = "A collection of gadgets for troubleshooting Kubernetes applications using eBPF";
+    homepage = "https://inspektor-gadget.io";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ kranuarg7 ];
```
Comment (wegank) [inline pkgs/applications/networking/cluster/kubectl-gadget/default.nix:28]: Typo:

```suggestion
    maintainers = with maintainers; [ kranurag7 ];
```

**Comment by kranurag7** [comment]
@wegank Can you please review this again subject to your time availability? I see the checks are passing now.

**Review by wegank — APPROVED** [review]

**Comment by wegank** [comment]
If there's no other review or if this PR isn't merged within the next 24 hours, ping me and I'll merge it.

**Comment by kranurag7** [comment]
@wegank Friendly ping for ⬆️

**Pull request merged**